src/cursor/cur_version.c

```diff
@@ -0,0 +1,552 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#define VERSION_CURSOR_METADATA_FORMAT WT_UNCHECKED_STRING(QQQQQQBBBB)
+/*
+ * __curversion_set_key --
+ *     WT_CURSOR->set_key implementation for version cursors.
+ */
+static void
+__curversion_set_key(WT_CURSOR *cursor, ...)
+{
+    WT_CURSOR *table_cursor;
+    WT_CURSOR_VERSION *version_cursor;
+    uint32_t flags;
+    va_list ap;
+
+    version_cursor = (WT_CURSOR_VERSION *)cursor;
+    table_cursor = version_cursor->table_cursor;
+    va_start(ap, cursor);
+    flags = table_cursor->flags;
+    if (F_ISSET(cursor, WT_CURSTD_RAW))
+        flags |= WT_CURSTD_RAW;
+    WT_IGNORE_RET(__wt_cursor_set_keyv(table_cursor, flags, ap));
+    va_end(ap);
+}
+
+/*
+ * __curversion_get_key --
+ *     WT_CURSOR->get_key implementation for version cursors.
+ */
+static int
+__curversion_get_key(WT_CURSOR *cursor, ...)
+{
+    WT_CURSOR *table_cursor;
+    WT_CURSOR_VERSION *version_cursor;
+    WT_DECL_RET;
+    uint32_t flags;
+    va_list *ap, _ap;
+
+    ap = &_ap;
+    version_cursor = (WT_CURSOR_VERSION *)cursor;
+    table_cursor = version_cursor->table_cursor;
+    va_start(*ap, cursor);
+    flags = table_cursor->flags;
+    if (F_ISSET(cursor, WT_CURSTD_RAW))
+        flags |= WT_CURSTD_RAW;
+    WT_ERR(__wt_cursor_get_keyv(table_cursor, flags, ap));
+
+err:
+    va_end(*ap);
+    return (ret);
+}
+
+/*
+ * __curversion_get_value --
+ *     WT_CURSOR->get_value implementation for version cursors.
+ */
+static int
+__curversion_get_value(WT_CURSOR *cursor, ...)
+{
+    WT_CURSOR *table_cursor;
+    WT_CURSOR_VERSION *version_cursor;
+    WT_DECL_RET;
+    va_list *ap, _ap;
+
+    ap = &_ap;
+    version_cursor = (WT_CURSOR_VERSION *)cursor;
+    table_cursor = version_cursor->table_cursor;
+    va_start(*ap, cursor);
+    if (F_ISSET(cursor, WT_CURSTD_RAW)) {
+        WT_ERR(__wt_cursor_get_valuev(cursor, "u", ap));
```

Comment:
@ddanderson @keithbostic I believe the root cause of the crash in windows is that the implementation of va_list is different in windows. If we pass the va_list directly to another function, we are effectively doing a va_copy in windows. I modify the code to pass a pointer instead and it is now passing test_cursor18 but test_cursor19 is still crashing. We never hit this issue before because previously we don't have a use case that extract variables from the va_list in two separate functions.

Reply:
Oh, ouch. Ummm, yeah, for some reason I knew that `va_lists` were local to a single function. Can we do an explicit `va_copy`? That is supposed to copy from the current `va_list` location, I believe, which would do what you need?

Reply:
I don't think that will work. If we pass va_list to the two functions directly, they are using the same state anyway. The ones passed to the functions are both copies.